csrc/host_ir/executor.h

```diff
@@ -53,7 +53,20 @@ class HostIrExecutor final : public OptInDispatch {
       Communicator* communicator = nullptr,
       HostIrExecutorParams = HostIrExecutorParams());
   std::vector<at::Tensor> runWithInput(
-      std::unordered_map<Val*, c10::IValue> val_to_IValue);
+      std::unordered_map<Val*, c10::IValue> val_to_IValue,
+      LaunchParams launch_params = LaunchParams());
```

Comment:
What's this for? IIUC, LaunchParams is per-segment, and I'm yet to understand why we need to pass it in to HostIrExecutor::runWithInput, which runs the whole host program spanning through multiple segments.

Reply:
> It is copied from what is currently implemented in MultiDeviceExecutor.

Good to know it's an old behavior copied over! 

Can you try your best to put some code comments? I bet LaunchParams being a global instead of a per-segment config will confuse many readers. It's good to understand how it's used today and the limitations.

Reply:
> Can you try your best to put some code comments? I bet LaunchParams being a global instead of a per-segment config will confuse many readers. It's good to understand how it's used today and the limitations.

Sure. I'm gonna add comments.
I agree that the current solution is not ideal. We should consider fixing it in a near future. I will keep this in mind.